tools/qc/fastq_utils/fastq_pre_barcodes.xml

```diff
@@ -0,0 +1,246 @@
+    <command detect_errors="exit_code"><![CDATA[
+    fastq_pre_barcodes --read1 '$read1' --outfile '$outfile1'
+
+    #if $read2:
+        --read2 '$read2'
+    #end if
+
+    #if $index1:
+        --index1 '$index1'
+    #end if
+
+    #if $index2:
+        --index2 '$index2'
+    #end if
+
+    #if $index3:
+        --index3 '$index3'
+    #end if
+
+    #if $phred_encoding:
+        --phred_encoding '$phred_encoding'
+    #end if
+
+    #if $min_qual:
+        --min_qual '$min_qual'
+    #end if
+
+    #if $outfile2:
+        --outfile2 '$outfile2'
+    #end if
+
+    #if $outfile3:
+        --outfile3 '$outfile3'
+    #end if
+
+    #if $interleaved:
+        --interleaved '$interleaved'
+    #end if
+
+    #if $umi_read:
+        --umi_read '$umi_read'
+    #end if
+
+    #if $umi_offset:
+        --umi_offset '$umi_offset'
+    #end if
+
+    #if $umi_size:
+        --umi_size '$umi_size'
+    #end if
+
+    #if $Cell_read:
+        --$Cell_read '$Cell_read'
+    #end if
+
+    #if $Cell_offset:
+        --Cell_offset '$Cell_offset'
+    #end if
+
+    #if $Cell_size:
+        --Cell_size '$Cell_size'
+    #end if
+
+    #if $sample_read: 
+        --sample_read '$sample_read'
+    #end if
+
+    #if $sample_offset:
+        --sample_offset '$sample_offset'
+    #end if
+
+    #if $sample_size:
+        --sample_size '$sample_size'
+    #end if
+
+    #if $read1_offset:
+        --read1_offset '$read1_offset'
+    #end if
+
+    #if $read1_size:
+        --read1_size '$read1_size'
+    #end if
+
+    #if $read2_offset:
+        --read2_offset '$read2_offset'
+    #end if
+
+    #if $read3_offset: 
+        --read3_offset '$read3_offset'
+    #end if
+
+    #if $use_10x:
+        '$use_10x'
+    #end if
+
+    #if $sam:
+        '$sam'
+    #end if
+
+    #if $brief: 
+        '$brief'
+    #elif $verbose:
+        '$verbose'
+    #end if
+    ]]></command>
+    <inputs>
+        <param name="verbose" label="Verbose" optional='true' value='false' argument="--verbose" type="boolean"  truevalue='--verbose' falsevalue='' checked='true' help="Increase level of messages printed to stderr"/>
+        <param name="brief" label="Brief" optional='true' value='true' argument="--brief" type="boolean"  truevalue='--brief' falsevalue='' checked='true' help="Decrease level of messages printed to stderr"/>
+        <param name="read1" label="Read1"  argument="--read1" type="data" format='?'  help="fastq (optional gzipped) file name"/>
+        <param name="read2" label="Read2"  argument="--read2" type="data" format='?'  help="fastq (optional gzipped) file name"/>
+        <param name="index1" label="Index1" argument="--index1" type="data" format='?'  help="fastq (optional gzipped) file name"/>
+        <param name="index2" label="Index2" argument="--index2" type="data" format='?'  help="fastq (optional gzipped) file name"/>
+        <param name="index3" label="Index3" argument="--index3" type="data" format='?'  help="fastq (optional gzipped) file name"/>
```

Comment:
We need format set here. I suspect `fastqsanger`.

Reply:
Done in 3896d3d2f77c47f9eca79bff42e7df7e12bbfdef.  I initially set format to `fastq,fastqsanger` because the param is supposed to accept a fastq file or a gzipped version, but this generates an error.  Finally set it to just `fastqsanger`, but needs an edit to include `fastq`.